pineapple-bundle/pineapple-core/src/main/java/sh/miles/pineapple/config/type/Configuration.java

```diff
@@ -0,0 +1,234 @@
+                config.setComments(path, field.getComments());
+            }
+
+            TypeAdapter<Object, Object> adapter = (TypeAdapter<Object, Object>) configManager.getTypeAdapter(ConfigType.get(field.getField()));
+            if (adapter == null) {
+                PineappleLib.getLogger().log(Level.WARNING, "Unable to find type adapter for field: {0} with path: {1} in {2}", objArr(field.getField().getName(), field.getPath(), this.file.getName()));
+                field.setHidden();
+                continue;
+            }
+
+
+            Object toSave = this.getValue(field.getField());
+            if (toSave == null) {
+                PineappleLib.getLogger().log(Level.WARNING, "Found null value while trying to save for field name: {0}, path: {1} in config {2}", objArr(field.getField().getName(), path, this.file.getName()));
+            }
+
+            Object existing = config.get(path);
+            if (existing != null) {
+                if (!replace) {
+                    continue;
+                }
+            }
+
+            if (existing instanceof ConfigurationSection section) {
+                existing = section.getValues(false);
+            }
+
+            Object value = adapter.write(toSave, existing, replace);
+            if (value == null) {
+                PineappleLib.getLogger().log(Level.WARNING, "Found null value from TypeAdapter: {0} while trying to save config {1}", objArr(adapter.getClass().getTypeName(), this.file.getName()));
+                field.setHidden();
+                continue;
+            }
+
+            if (mapOnly) {
+                Map<String, Object> saving = (Map<String, Object>) value;
+                for (Map.Entry<String, Object> entry : saving.entrySet()) {
+                    config.set(entry.getKey(), entry.getValue());
+                }
+                field.setHidden();
+                break;
+            }
+
+
+            config.set(path, value);
+            field.setHidden();
+        }
+
+        try {
+            config.save(this.file);
+        } catch (IOException ex) {
+            PineappleLib.getLogger().log(Level.WARNING, ex, () -> "Unable to save config %s".formatted(this.file.getName()));
+        }
+        return this;
+    }
+
+    private void setField(Field field, Object value) {
+        this.fieldModifier.setField(field, value, instance);
+    }
+
+    private Object getValue(Field field) {
+        return this.fieldModifier.getValue(field, instance);
+    }
+
+
+    private boolean setupFile() {
+        try {
+            if (this.file.isDirectory()) {
+                return false;
+            }
+
+            if (!this.file.getParentFile().exists() && !this.file.getParentFile().mkdirs()) {
+                return false;
+            }
+
+            if (!this.file.exists()) {
+                return this.file.createNewFile();
+            }
+
+            return true;
+        } catch (IOException ex) {
+            return false;
```

Comment:
instead of returning false here we should probably throw a runtime exception its genuinely useful to know when READ/WRITE permissions aren't correct. Leaves less mystery imho. Idk if this is precedent elsewhere in the entire project, but those places should probably be converted too eventually.

Reply:
If you look where this is used it does say there is an issue with loading it, this doesnt do anything with the error by design. I can return more data with the boolean of why if you want

Reply:
Or i can print the error on logger

Reply:
log the exception details to console in this method please. The later log you use is fine as well.